### PR TITLE
Update select poll logic to support eventlet.

### LINKED ISSUE #1753: WSClient.update() fails on select.poll() when running in eventlet on Linux
**What happened (please include outputs or screenshots)**:

WSClient.update() was modified by PR https://github.com/kubernetes-client/python-base/pull/268 to use select.poll() instead of select.select, which breaks when running under eventlet on Linux:

  File "/usr/local/lib/python3.8/site-packages/kubernetes/stream/ws_client.py", line 182, in update
    poll = select.poll()
AttributeError: module 'select' has no attribute 'poll'
module 'select' has no attribute 'poll'

eventlet monkey_patches the select module and does not support the poll() method

Adding a check on getattr(select, "poll", None) is None to the if statement and reverting to select.select when there is no poll attribute will fix the problem.

I'll push a PR

**What you expected to happen**:

No exception should occur

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (`kubectl version`): 1.19, 1.21
- OS (e.g., MacOS 10.13.6): CentOS 7
- Python version (`python --version`)  3.8
- Python client version (`pip list | grep kubernetes`)  23.3.0

### PR DESCRIPTION
**Pull request by bobh66**
#### What type of PR is this?
/kind bug
/kind regression

#### What this PR does / why we need it:
Fixes exception in WSClient.update() caused by select not having a poll() attribute when running under eventlet. Introduced by PR https://github.com/kubernetes-client/python-base/pull/268

#### Which issue(s) this PR fixes:
Fixes #1753 

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
```

### CODE REVIEW

File: kubernetes/base/stream/ws_client.py [hunk 1]
```diff
@@ -179,7 +179,9 @@ def update(self, timeout=0):
         #                   efficient as epoll. Will work for fd numbers above 1024.
         # select.epoll()  - newest and most efficient way of polling.
         #                   However, only works on linux.
-        if sys.platform.startswith('linux') or sys.platform in ['darwin']:
+        #                   Will not work on Linux when eventlet is being used.
+        if ((sys.platform.startswith('linux') and getattr(select, "poll", None) is not None) or
```
Comment (yliaog) [inline kubernetes/base/stream/ws_client.py:183]: eventlet supports epoll, does it make sense to use epoll

Comment (bobh66) [reply kubernetes/base/stream/ws_client.py:183]: If that's a better solution I'm happy to rework the PR.  I'm not very familiar with select and poll but if epoll is equivalent I'll update the change.

Comment (Akasurde) [inline kubernetes/base/stream/ws_client.py:183]: You can take inspiration from https://github.com/python/cpython/pull/21257 and https://github.com/ansible/ansible/pull/70352

**Comment by roycaihw** [comment]
Closing due to inactivity. 
/close

**Pull request closed without merging**